src/index.js

```diff
@@ -98,7 +99,47 @@ export default (Bookshelf, options = {}) => {
         // Initialize an instance of the current model and clone the initial query
         internals.model =
             this.constructor.forge().query((qb) => _assign(qb, this.query().clone()));
+        /**
+         * Build a query for single filtering dependency object
+         * @param   value {object}
+         * @param   relationHash {object}
+         */
+        internals.buildObjectLikeFilterValue = (value, relationHash) => {
 
+            if (!_isEmpty(value)){
+                _forEach(value, (_, typeKey) => {
+                    // Add relations to the relationHash
+                    internals.buildDependenciesHelper(typeKey, relationHash);
+                });
+            }
+        };
+        /**
+         * Build a query for array of relational dependencies of filtering
+         * @param   filterList {array}
+         * @param   relationHash {object}
+         */
+        internals.buildArrayLikeFilterValue = (filterList, relationHash) => {
```

Comment:
Can we rename this to be more specific that it is only for `or` filters?

Reply:
Done